### PR TITLE
gRPC SDK Authorization Policy proto.

### PR DESCRIPTION
**Pull request by ashithasantosh**
Defines SDK Authorization Policy proto. This is required for gRPC OSS Authorization effort, allowing non-xDS users to provide an authorization policy.

### CODE REVIEW

File: src/proto/grpc/auth/v1/BUILD [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# limitations under the License.
+
+licenses(["notice"])  # Apache v2
+
+load("@rules_proto//proto:defs.bzl", "proto_library")
+
+proto_library(
+    name = "sdk_auth_policy_proto",
+    srcs = [
+        "sdk_auth_policy.proto",
```
Comment (jiangtaoli2016) [inline src/proto/grpc/auth/v1/BUILD:22]: I would name this proto file: sdk_authz_policy.proto
Please also change the corresponding upb files.

File: src/proto/grpc/auth/v1/sdk_auth_policy.proto [hunk 2]
```diff
@@ -0,0 +1,101 @@
+// Copyright 2018 The gRPC Authors
```
Comment (jiangtaoli2016) [inline src/proto/grpc/auth/v1/sdk_auth_policy.proto:1]: nit: 2021

File: src/proto/grpc/auth/v1/sdk_auth_policy.proto [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  Peer source = 2;
+
+  // Optional. If not set, no checks will be performed against the request. An empty
+  // rule is always matched (i.e., both source and request are empty).
+  Request request = 3;
+}
+
+// AuthorizationPolicy defines which principals are permitted to access which
+// resource. Resources are RPC methods scoped by services.
+message AuthorizationPolicy {
```
Comment (jiangtaoli2016) [inline src/proto/grpc/auth/v1/sdk_auth_policy.proto:87]: I would add an example here

```
// In the following policy example, a peer identity from ["admin1", "admin2", "admin3"] 
// is authorized to access any RPC methods in pkg.service, and peer identity "dev" is 
// authorized to access the "foo" and "bar" RPC methods.
//
// name: example-policy
// allow_rules:
// - name: admin-access
//   source:
//     principals:
//     - "spiffe://foo.com/sa/admin1"
//     - "spiffe://foo.com/sa/admin2"
//     - "spiffe://foo.com/sa/admin3"
//   request:
//     paths: ["/pkg.service/*"]
// - name: dev-access
//   source:
//     principals: ["dev"]
//   request:
//     paths: ["/pkg.service/foo", "/pkg.service/bar"]
```

**Review by jiangtaoli2016 — COMMENTED** [review]
Thanks Ashitha for the PR. Minor cosmetics changes.

**Review by jiangtaoli2016 — APPROVED** [review]
Thanks Ashitha!

**Comment by ashithasantosh** [comment]
Thanks Jiangtao for the review!:)

**Comment by markdroth** [comment]
Please remove the "sdk_" prefix from the filename.  We don't use that on any of our other files, so there's no reason we should do it here.

Also, the authoritative source for this proto file should be in the https://github.com/grpc/grpc-proto repo.

**Comment by ashithasantosh** [comment]
> Please remove the "sdk_" prefix from the filename. We don't use that on any of our other files, so there's no reason we should do it here.
> 
> Also, the authoritative source for this proto file should be in the https://github.com/grpc/grpc-proto repo.

Done. Removed sdk_ prefix. Will make the same changes in grpc-proto repo.

**Review by markdroth — APPROVED** [review]

**Pull request merged**